pkg/services/mssqlfg/database_pair_fe_primary_provision.go

```diff
@@ -0,0 +1,167 @@
+package mssqlfg
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/open-service-broker-azure/pkg/service"
+	uuid "github.com/satori/go.uuid"
+)
+
+func (d *databasePairFePrimaryManager) GetProvisioner(
+	service.Plan,
+) (service.Provisioner, error) {
+	return service.NewProvisioner(
+		service.NewProvisioningStep("preProvision", d.preProvision),
+		service.NewProvisioningStep(
+			"checkNameAvailability",
+			d.checkNameAvailability,
+		),
+		service.NewProvisioningStep("getPriDatabase", d.getPriDatabase),
+		service.NewProvisioningStep("deployPriARMTemplate", d.deployPriARMTemplate),
+		service.NewProvisioningStep("deployFgARMTemplate", d.deployFgARMTemplate),
+		service.NewProvisioningStep("deploySecARMTemplate", d.deploySecARMTemplate),
+	)
+}
```

Comment:
In `azure-sql-12-0-dr-database-pair-from-existing-primary`, what if the name of existing database is A and the user provide B in provisioning parameter `database`?